### PR TITLE
Client telemetry service integration and enabling on CTL

### PR DESCRIPTION
**Pull request by simplynaveen20**
This PR covers
1. SDK integrating with client telemetry endpoint
2. Currently we will leverage system variable for providing the endpoint 
3. Enabling it for our daily CTL run

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java [hunk 1]
```diff
@@ -443,7 +443,7 @@ public void init(CosmosClientMetadataCachesSnapshot metadataCachesSnapshot, Func
             clientTelemetry = new ClientTelemetry(null, UUID.randomUUID().toString(),
                 ManagementFactory.getRuntimeMXBean().getName(), userAgentContainer.getUserAgent(),
                 connectionPolicy.getConnectionMode(), globalEndpointManager.getLatestDatabaseAccount().getId(),
-                null, null, httpClient(), connectionPolicy.isClientTelemetryEnabled());
+                null, null, reactorHttpClient, connectionPolicy.isClientTelemetryEnabled(), this);
```
Comment (mbhaskar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:446]: nit: this.reactorHttpClient

**Review by mbhaskar — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java [hunk 2]
```diff
@@ -123,26 +145,75 @@ public void close() {
...
+                    return Mono.empty();
+                }
+
+                if (!Configs.isClientTelemetryEnabled(this.isClientTelemetryEnabled)) {
+                    logger.trace("client telemetry not enabled");
+                    return Mono.empty();
+                }
+                readHistogram();
+                try {
+                    String endpoint = Configs.getClientTelemetryEndpoint();
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java:160]: I would cache `Configs.getClientTelemetryEndpoint()` value in `ClientTelemetry`. to not pay the additional overhead of `System.getProperty(CLIENT_TELEMETRY_ENDPOINT)` each time. 

do we expect the value of `getClientTelemetryEndpoint` change at runtime? is that why this is not set as field in ClientTelemetry, or any other reason?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java [hunk 3]
```diff
@@ -254,6 +254,9 @@
         public static final String IS_BATCH_ORDERED = "x-ms-cosmos-batch-ordered";
         public static final String IS_BATCH_REQUEST = "x-ms-cosmos-is-batch-request";
         public static final String SHOULD_BATCH_CONTINUE_ON_ERROR = "x-ms-cosmos-batch-continue-on-error";
+
+        //Client telemetry header
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java:258]: nit: please change to be of consistent formatting with other comments in this file.
```suggestion
        // Client telemetry header
```

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java [hunk 4]
```diff
@@ -88,6 +88,7 @@ public AsyncCtlWorkload(Configuration cfg) {
             .endpoint(cfg.getServiceEndpoint())
             .key(cfg.getMasterKey())
             .consistencyLevel(cfg.getConsistencyLevel())
+            .clientTelemetryEnabled(true)
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java:91]: something to consider for future (not a blocking question).

can we measure in CTL the perf diff latency P99 and throughput when enabling client telemetry is enabled comparing to disabled.

**Review by moderakh — APPROVED** [review]
LGTM. minor questions, comments.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java:160]: That was the intention to keep clientTelemetryEnabled and clientTelemetryEnpoint changeable on the runtime. This is one call in 10 min , so not an issue. But your point is more applicable to TracerProvider and CosmosPageFlux where we are calling Configs.isClientTelemetryEnabled() on every response. However during the perf testing we did not overserve any perf hit. We should talk offline. Also I noticed in gateway call (http call ) we are calling config methods on every call including query plan

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java (hunk 4 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java:91]: sure

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java:160]: Resolving as per offline conversation , this need to be changed all across sdk at once

**Pull request merged**